pkg/sql/set_var.go

```diff
@@ -61,6 +62,15 @@ func (p *planner) SetVar(ctx context.Context, n *tree.SetVar) (planNode, error)
 	if err != nil {
 		return nil, err
 	}
+	if _, ok := settings.Lookup(name, settings.LookupForLocalAccess, p.ExecCfg().Codec.ForSystemTenant()); ok {
```

Comment:
hmm, actually, why is the codec for system tenant?

Reply:
`p.ExecCfg().Codec.ForSystemTenant()` returns a bool value indicating if it's on the system tenant or not.